### ISSUE TITLE
`--ignore` option can not handle absolute path?

### ISSUE THREAD

**Issue by ybiquitous** [igorshubovych/markdownlint-cli#31]
Hi! New feature `--ignore` option from `v0.7.0` is very awesome!

I found that this option behavior is different from ESLint `--ignore-pattern` option, so is this behavior a bug?

### OK

```sh
# relative path
eslint --ignore-pattern a.js "*.js"

# absolute path
eslint --ignore-pattern a.js "$(pwd)/*.js"

# relative path
markdownlint --ignore a.md "*.md"
```

### NG

```sh
# absolute path
markdownlint --ignore a.md "$(pwd)/*.md"
```

I created Gist for reproduction: 
https://gist.github.com/ybiquitous/3c7680dc9020455013c59fd999ab6fcd

Thanks.

**Comment by DavidAnson** [comment]
`ignore` works by doing a simple string compare after globbing. I think the failure case above would pass if the absolute path were used for `a.md` as well.

**Comment by ybiquitous** [comment]
Thanks quick response!

> `ignore` works by doing a simple string compare after globbing. I think the failure case above would pass if the absolute path were used for `a.md` as well.

This case works well on manual command-line.
But when using [`lint-staged`](https://npm.im/lint-staged), `--ignore` option doesn't work as I expect.

Please Update `package.json` in [that Gist](https://gist.github.com/ybiquitous/3c7680dc9020455013c59fd999ab6fcd#file-package-json).

Then update `a.md` manually.

```diff
diff --git a/a.md b/a.md
index 08635d0..bbd0857 100644
--- a/a.md
+++ b/a.md
@@ -1 +1,2 @@
 #a
+
```

And then run `npm run precommit` command.

```
$ npm run precommit

> @ precommit /private/tmp/mdl
> lint-staged

 ❯ Running tasks for *.md
   ✖ markdownlint --ignore a.md
     → /private/tmp/mdl/a.md: 3: MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
✖ markdownlint --ignore a.md found some errors. Please fix them and try committing again.

/private/tmp/mdl/a.md: 1: MD018/no-missing-space-atx No space after hash on atx style header [Context: "#a"]
/private/tmp/mdl/a.md: 1: MD041/first-line-h1 First line in file should be a top level header [Context: "#a"]
/private/tmp/mdl/a.md: 3: MD012/no-multiple-blanks Multiple consecutive blank lines [Expected: 1; Actual: 2]
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ precommit: `lint-staged`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @ precommit script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/***/.npm/_logs/2018-02-05T07_50_54_139Z-debug.log
```

Maybe I think that this tool (`lint-staged`) handle `a.md` as absolute path... 😓

**Comment by ybiquitous** [comment]
I tried to handle input files as absolute path:

```diff
diff --git a/markdownlint.js b/markdownlint.js
index 680f296..bb976df 100755
--- a/markdownlint.js
+++ b/markdownlint.js
@@ -54,7 +54,7 @@ function prepareFileList(files) {
     }
     return file;
   });
-  return flatten(files);
+  return flatten(files).map(file => path.resolve(file));
 }
 
 function lint(lintFiles, config) {
diff --git a/test/test.js b/test/test.js
index 9161f1d..2704dc6 100644
--- a/test/test.js
+++ b/test/test.js
@@ -1,5 +1,6 @@
 'use strict';
 
+import path from 'path';
 import test from 'ava';
 import execa from 'execa';
 
@@ -89,6 +90,13 @@ test('glob linting with failing files passes when failures ignored by glob', asy
   t.true(result.stderr === '');
 });
 
+test('glob linting with failing files passes when failures ignored by glob', async t => {
+  const result = await execa('../markdownlint.js',
+    ['--config', 'test-config.json', path.resolve('incorrect.md'), '--ignore', '**/incorrect.md']);
+  t.true(result.stdout === '');
+  t.true(result.stderr === '');
+});
+
 test('glob linting with failing files passes when everything ignored by glob', async t => {
   const result = await execa('../markdownlint.js',
     ['--config', 'test-config.json', '**/*.md', '--ignore', '**/*']);

```

But test failed:

```
$ npm test

   12 passed
   1 failed


   1. linting results are sorted by file/line/names/description

  t.true(err.stderr === expected)
             |          |
             |          "incorrect.md: 1: MD002/first-header-h1 First header should be a top level header [Expected: h1; Actual: h2]\nincorrect.md: 1: MD022/blanks-around-headers Headers should be surrounded by blank lines [Context: \"## header 2\"]\nincorrect.md: 1: MD041/first-line-h1 First line in file should be a top level header [Context: \"## header 2\"]\nincorrect.md: 2: MD022/blanks-around-headers Headers should be surrounded by blank lines [Context: \"# header\"]\nincorrect.md: 4: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\nincorrect.md: 10: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\nincorrect.md: 16: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\nincorrect.md: 23: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\n"
             "/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 1: MD002/first-header-h1 First header should be a top level header [Expected: h1; Actual: h2]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 1: MD022/blanks-around-headers Headers should be surrounded by blank lines [Context: \"## header 2\"]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 1: MD041/first-line-h1 First line in file should be a top level header [Context: \"## header 2\"]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 2: MD022/blanks-around-headers Headers should be surrounded by blank lines [Context: \"# header\"]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 4: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 10: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 16: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 23: MD014/commands-show-output Dollar signs used before commands without showing output [Context: \"$ code\"]\n"

      "incorrect.md: 1: MD002/first-header-h1 First header should be a top level header [Expected: h1; Actual: h2]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 1: MD022/blanks-around-headers Headers should be surrounded by blank lines [Context: /"## header 2/"]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 1: MD041/first-line-h1 First line in file should be a top level header [Context: /"## header 2/"]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 2: MD022/blanks-around-headers Headers should be surrounded by blank lines [Context: /"# header/"]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 4: MD014/commands-show-output Dollar signs used before commands without showing output [Context: /"$ code/"]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 10: MD014/commands-show-output Dollar signs used before commands without showing output [Context: /"$ code/"]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 16: MD014/commands-show-output Dollar signs used before commands without showing output [Context: /"$ code/"]/n/Users/***/git/ybiquitous/markdownlint-cli/test/incorrect.md: 23: MD014/commands-show-output Dollar signs used before commands without showing output [Context: /"$ code/"]/n"
        _callee16$ (test.js:154:5)
        tryCatch (/Users/***/git/ybiquitous/markdownlint-cli/node_modules/regenerator-runtime/runtime.js:62:40)
        Generator.invoke [as _invoke] (/Users/***/git/ybiquitous/markdownlint-cli/node_modules/regenerator-runtime/runtime.js:296:22)
        Generator.prototype.(anonymous function) [as throw] (/Users/***/git/ybiquitous/markdownlint-cli/node_modules/regenerator-runtime/runtime.js:114:21)
        <anonymous>
        process._tickCallback (internal/process/next_tick.js:160:7)
npm ERR! Test failed.  See above for more details.
```

**Comment by ybiquitous** [comment]
Surely relative path saves displayed console area than absolute path, such as:

```
README.md: 1: MD013/line-length Line length [Expected: 80; Actual: 132]
/path/to/project/README.md: 1: MD013/line-length Line length [Expected: 80; Actual: 132]
```

But I think that absolute path is more integrable with 3rd-party lint tools (e.g. `lint-staged`) than relative path.

What do you think about this?

**Comment by DavidAnson** [comment]
`lint-staged` is documented to pass absolute paths, so for now you can probably use the same for `ignore` and not be blocked.

Your code change above could probably be made to work by having `prepareFileList` return the original and resolved path, excluding based on resolved, then using original for display.

I can look at making this change later this week.

Thanks for the very detailed issue and notes!!

**Comment by ybiquitous** [comment]
Thanks for feedback! I will cooperate with anything I can do! 😊👍

**Comment by ybiquitous** [comment]
@DavidAnson Thank you so much for very quick fix!!! 🙇🙇🙇

**Comment by DavidAnson** [comment]
:) I’ll probably publish a new version tomorrow.
